src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs

```diff
@@ -161,5 +161,10 @@ public sealed override void Visit(IValueVisitor visitor)
         {
             visitor.Visit(this);
         }
+
+        public virtual void UpdateField(NamedValue newNamedValue)
```

Comment:
>UpdateField(NamedValue newNamedValue) [](http://example.com/codeflow?start=28&length=37)

We should also have an overload:
  Update(string name, FormulaValue newValue)

Note that only 1 overload should be virtual, the others should chain.